packages/plugin-ext/src/main/browser/commands.ts

```diff
@@ -100,3 +102,22 @@ class OpenNewTabDialog extends AbstractDialog<string> {
         this.open();
     }
 }
+
+@injectable()
+export class DiffCommandHandler {
```

Comment:
Isn't it the same as [`file.compare`](https://github.com/theia-ide/theia/blob/60206635557685f049e1334362d491966f519171/packages/workspace/src/browser/workspace-commands.ts#L107-L111)? If so, please do not duplicate existing functionality.

By the way, your code does not handle when the URI is pointing a directory.